nixos/modules/services/misc/evremap.nix

```diff
@@ -117,7 +129,9 @@ in
       description = "evremap - keyboard input remapper";
       wantedBy = [ "multi-user.target" ];
 
-      script = "${lib.getExe pkgs.evremap} remap ${format.generate "evremap.toml" cfg.settings}";
+      script = "${lib.getExe pkgs.evremap} remap ${
+        format.generate "evremap.toml" (lib.attrsets.filterAttrs (n: v: v != null) cfg.settings)
```

Comment:
This is getting a bit lengthy so maayyyyyybe consider moving it out

Reply:
sure, how about like this:
```nix
let
  cfg = config.services.evremap;
  format = pkgs.formats.toml { };

  settings = lib.attrsets.filterAttrs (n: v: v != null) cfg.settings;

...

    script = "${lib.getExe pkgs.evremap} remap ${format.generate "evremap.toml" settings}";
```

Reply:
Oh I intended to mean to move the generated `evremap.toml` derivation out, and name it something like `configFile`, and letting the script where it is

Reply:
that does make a lot more sense than my assumption you were talking about my changes only

I think this should be satisfying then?

```nix
let
  cfg = config.services.evremap;
  format = pkgs.formats.toml { };

  settings = lib.attrsets.filterAttrs (n: v: v != null) cfg.settings;
  configFile = format.generate "evremap.toml" settings;

...

  script = "${lib.getExe pkgs.evremap} remap ${configFile}";
```

I can push this if you think its fitting.

What should we do about the description of `phys` and the reboot thing?

Reply:
Yeah, looks good - the description for `phys` is OK as well, maybe could change the wording a bit

Reply:
Great, I'll push in the evening today